src/SonarScanner.MSBuild.Tasks/WriteProjectInfoFile.cs

```diff
@@ -344,12 +336,28 @@ private bool TryGetSettingValue(ITaskItem taskItem, out string metadataValue)
 
             Log.LogMessage(Resources.WPIF_GeneratingRandomGuid, FullProjectPath, generatedGuid);
 
-            //Generating a new guid for projects without one
+            // Generating a new guid for projects without one.
             return generatedGuid;
 
-            bool ArePathEquals(string filePath, FileInfo file) =>
-                filePath != null &&
-                new FileInfo(filePath).FullName.Equals(file.FullName, StringComparison.OrdinalIgnoreCase);
+            bool ArePathEquals(string filePath, FileInfo file)
+            {
+                if (filePath == null)
+                {
+                    return false;
+                }
+                else
+                {
+                    try
+                    {
+                        var fileInfo = new FileInfo(filePath);
+                        return fileInfo.FullName.Equals(file.FullName, StringComparison.OrdinalIgnoreCase);
```

Comment:
this should be case sensitive on Linux, right?

I'm not sure in general of our state for Windows vs Linux for the scanner, the problem might be bigger than this specific PR... maybe open a trello card to evaluate where we are regarding this?

Reply:
or actually you can do a quick search in the codebase for fileInfo usage and see if we do this sort of comparison in other places and directly open an issue to make it Linux compliant

it should be fast to do the check and decide if it's just in this place or in more, and maybe put an item for the Scanner agenda to discuss this Linux compatibility

Reply:
Based on [this ](https://github.com/SonarSource/sonar-scanner-msbuild/blob/37222dca4af5f5e844c3ae1444d684467cf8a40f/src/SonarScanner.MSBuild.Common/FileInfoEqualityComparer.cs#L29)comment and behavior I would say we are safe. WDYT?

Reply:
I don't really understand that comment. It doesn't give a link to some documentation, we cannot assume that the .NET Core behavior is the same in .NET 6.

It's good that you found that, though - why not use `FileInfoEqualityComparer` instead of manually comparing the path? And we can have a ticket to confirm that assumption from `FileInfoEqualityComparer`

Reply:
I changed to `FileInfoEqualityComparer ` and added a card to the scanner board to investigate the behavior.